src/components/lists/components.lists.todolist.jsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from "react";
+import "./components.lists.todolist.css";
+import Form from "../form/components.form.form";
+import logo from "./bin.png";
+
+//basic function for each todo item in the list
+function Todo({ todo, index, finishTodo, deleteTodo }) {
+  return (
+    <li>
+      {/* if clicked, move the item to the other list */}
+      <span onClick={() => finishTodo(index, todo.text)}>{todo.text}</span>
+      {/* if clicked, delete the item */}
+      <div className="button">
+        <img
+          src={logo}
+          alt="img"
+          onClick={() => deleteTodo(index, todo.text)}
+          align="right"
+        />
+      </div>
+    </li>
+  );
+}
+
+//function of whole todolist
+export function TodoList() {
+  //use of react hook to create todolist and finishedlist
+  const [listOfTodo, setTodo] = useState([]);
+  const [listOfDone, setDone] = useState([]);
+  //if called, add the item to the todolist
+  const addTodo = (text) => {
+    const tempTodo = [...listOfTodo, { text }];
+    setTodo(tempTodo);
+  };
+  //if called, move the item to the finishedlist
+  const finishTodo = (index, text) => {
+    addDone(text);
+    deleteTodo(index);
+  };
+  //if called, delete the item from the todolist
+  const deleteTodo = (index) => {
+    const tempTodo = [...listOfTodo];
+    tempTodo.splice(index, 1);
+    setTodo(tempTodo);
+  };
```

Comment:
```typescript
interface todoItem{
    text: string,
    id: number,
    isDone: boolean
}
```
이런식으로 todo item을 정의 했다면
```suggestion
  const deleteTodo = (index) => {
    setTodo(listOfTodo.filter((todo) => todo.id !== index);
  };
```
이런 방식으로 index에 의존하여 array를 조작하는 방법을 사용하지 않을수 있었을 것 같습니다!